src/main/java/com/example/demo/service/BookServiceImpl.java

```diff
@@ -9,16 +13,27 @@
 @Service
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
-
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto findBookById(Long id) {
+        Book book = bookRepository.findBookById(id).orElseThrow(
+                () -> new EntityNotFoundException(
+                        "Can`t find book with id: " + id));
+        return bookMapper.toDto(book);
+    }
 
     @Override
-    public Book save(final Book product) {
-        return bookRepository.save(product);
+    public BookDto save(CreateBookRequestDto bookDto) {
+        Book book = bookMapper.toModel(bookDto);
+        return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
```

Comment:
Is it possible to set up your mapper to map a list of entities to a list of entities? Instead of using streams and map each item one by one?